px-map-behavior-vector-layer.es6.js

```diff
@@ -0,0 +1,147 @@
+(function() {
+  'use strict';
+
+  /****************************************************************************
+   * BEHAVIORS
+   ****************************************************************************/
+
+  /* Ensures the behavior namespace is created */
+  window.PxMapBehavior = (window.PxMapBehavior || {});
+
+  /**
+   *
+   * @polymerBehavior PxMapBehavior.VectorLayer
+   */
+  PxMapBehavior.VectorLayerImpl = {
+    properties: {
+      /**
+       * An object that will contain the geojson data to display:
+       *
+       * @type {Object}
+       */
+      data: {
+        type: Object,
+        observer: 'shouldUpdateInst',
+        reflectToAttribute: true
+      },
+
+      /**
+       * An object that can specify the style
+       *
+       * @type {Object}
+       */
+      featureStyle: {
+        type: Object,
+        value: null
+      },
+
+      /**
+       * A boolean that if set will show a popover containing all the feature properties when the feature is clicked
+       *
+       * @type {Boolean}
+       */
+      showFeatureProperties: {
+        type: Boolean,
+        value: false
+      },
+
+      /**
+       * A boolean that if set will allow a blank vector layer to be created
+       *
+       * @type {Boolean}
+       */
+      blankLayer: {
+        type: Boolean,
+        value: false
+      }
+    },
+
+    canAddInst() {
+      return ((typeof this.data === 'object') && Object.keys(this.data).length) || this.blankLayer;
+    },
+
+    createInst(options) {
+      if(!options.featureStyle) options.featureStyle = {};
+
+      let geojsonLayer = L.geoJson(options.data, {
+        pointToLayer: (feature, latlng) => {
+          const featureProperties = feature.properties.style || {};
+          const attributeProperties = options.featureStyle;
+          const style = this._getStyle(feature, featureProperties, attributeProperties);
+
+          return new L.CircleMarker(latlng, style);
+        },
+
+        onEachFeature: (feature, layer) => {
+          if (!this.showFeatureProperties) return;
+
+          const popupData = Object.keys(feature.properties).reduce((accum, key) => {
+            if (key === 'style'){ return accum } // Filters out `feature.properties.style`.
+
+            if(feature.properties[key] && feature.properties[key] !== 'unset') {
```

Comment:
Think we can fold all this logic into one if at the top that returns `accum` directly, and let the happy path be returning `accum` the new key.